interfaces/udev/spec.go

```diff
@@ -93,6 +108,14 @@ func (c byTagAndSnippet) Less(i, j int) bool {
 
 // Snippets returns a copy of all the snippets added so far.
 func (spec *Specification) Snippets() (result []string) {
+	// If one of the interfaces controls it's own device cgroup, then
+	// we don't want to enforce a device cgroup, which is only turned on if
+	// there are udev rules, and as such we don't want to generate any udev
+	// rules
+
+	if spec.ControlsDeviceCgroup() {
+		return nil
+	}
 	entries := make([]entry, len(spec.entries))
 	copy(entries, spec.entries)
 	sort.Sort(byTagAndSnippet(entries))
```

Comment:
It would be good to add some tests to udev/spec_test.go (eg, like we did for suppressHomeIx).

Reply:
Done